### PR TITLE
Bump up babelfishpg_tsql extension version to 1.3.0

### PR DESCRIPTION
**Pull request by HarshLunagariya**
### Description

This commit bumps up babelfishpg_tsql extension version to 1.3.0 and adds upgrade script for 1.2.1 to 1.3.0 upgrade path.

### Test Scenarios Covered ###
* **Use case based -**
NA

* **Boundary conditions -**
NA

* **Arbitrary inputs -**
NA

* **Negative test cases -**
NA

* **Minor version upgrade tests -**
NA

* **Major version upgrade tests -**
NA

* **Performance tests -**
NA

* **Tooling impact -**
NA

* **Client tests -**
NA

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by rishabhtanwar29 — DISMISSED** [review]

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--1.2.1--1.3.0.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- complain if script is sourced in psql, rather than via ALTER EXTENSION
+\echo Use "ALTER EXTENSION ""babelfishpg_tsql"" UPDATE TO '1.3.0'" to load this file. \quit
+ 
+SELECT set_config('search_path', 'sys, '||current_setting('search_path'), false);
+
```
Comment (kuntalghosh) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--1.2.1--1.3.0.sql:5]: Shouldn't we make the changes for babelfish_drop_deprecated_object procedure?

Comment (HarshLunagariya) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--1.2.1--1.3.0.sql:5]: It is not needed for current changes.

**Review by kuntalghosh — APPROVED** [review]

**Review by kuntalghosh — COMMENTED** [review]
Could you please check if there's any MVU path from 1x dev to any other 2x dev? If I merge this, will it cause a MVU failure in 2x dev as 1.3.0-2.* scripts doesn't exist?

**Review by rishabhtanwar29 — APPROVED** [review]

**Comment by 2jungkook** [comment]
We should merge this after https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/863

**Review by 2jungkook — APPROVED** [review]

**Pull request merged**